src/sql/src/plan/lowering.rs

```diff
@@ -792,17 +792,22 @@ impl HirScalarExpr {
                                 .project((0..inner_arity).chain(Some(then_arity)).collect());
 
                             // Restrict to records not satisfying `cond_expr` and apply `els` as a map.
-                            let mut else_inner = get_inner.filter(vec![SS::CallBinary {
-                                func: mz_expr::BinaryFunc::Or,
-                                expr1: Box::new(SS::CallBinary {
-                                    func: mz_expr::BinaryFunc::Eq,
-                                    expr1: Box::new(cond_expr.clone()),
-                                    expr2: Box::new(SS::literal_ok(Datum::False, ScalarType::Bool)),
-                                }),
-                                expr2: Box::new(SS::CallUnary {
-                                    func: mz_expr::UnaryFunc::IsNull(mz_expr::func::IsNull),
-                                    expr: Box::new(cond_expr.clone()),
-                                }),
+                            let mut else_inner = get_inner.filter(vec![SS::CallVariadic {
+                                func: mz_expr::VariadicFunc::Or,
+                                exprs: vec![
+                                    SS::CallBinary {
+                                        func: mz_expr::BinaryFunc::Eq,
+                                        expr1: Box::new(cond_expr.clone()),
+                                        expr2: Box::new(SS::literal_ok(
+                                            Datum::False,
+                                            ScalarType::Bool,
+                                        )),
```

Comment:
opportunity to use the existing `MirScalarExpr::call_binary` and the new `MirScalarExpr::call_is_null` function here.

Reply:
Done